docs/book/tutorial/getting-started/beginner_tutorial.md

```diff
@@ -0,0 +1,492 @@
+in a common package to share them. Time to get our hands dirty and install the package.
+
+First create an empty folder called `prooph_tutorial` and `cd` into it. 
+
+### Require prooph/common
+
+`$ composer require prooph/common`
+
+This command will run composer that generates a fresh `composer.json` for us and adds `prooph/common`
+as first package to our new project. Let's check what we've installed.
+
+### The first message
+
+Create a file called `hello_world.php` with the following content and run it with `php`:
+
+```php
+<?php
+//All prooph components use strict types enabled
+declare(strict_types=1);
+
+namespace ProophTutorial;
+
+use Prooph\Common\Messaging\Command;
+use Prooph\Common\Messaging\PayloadConstructable;
+use Prooph\Common\Messaging\PayloadTrait;
+
+//Require composer's autoloader
+require 'vendor/autoload.php';
+
+//Our first message
+final class SayHello extends Command implements PayloadConstructable
+{
+    use PayloadTrait;
+
+    public function to(): string
+    {
+        return $this->payload['to'];
+    }
+}
+
+$sayHello = new SayHello(['to' => 'World']);
+
+echo 'Hello ' . $sayHello->to();
+
+//Hello World
+
+```
+
+In the script above we've created our first message of message type `command`.
+We've also used `Prooph\Common\Messaging\PayloadTrait` in conjunction with the `Prooph\Common\Messaging\PayloadConstructable` interface
+to be able to instantiate our command with a `payload` - a simple array - and get access to it using
+`$this->payload` within the message. 
+While this is a very easy and fast way to create message classes it is completely optional.
+The most important thing to note here is that `Prooph\Common\Messaging\Command` implements `Prooph\Common\Messaging\Message`
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace Prooph\Common\Messaging;
+
+use DateTimeImmutable;
+use Ramsey\Uuid\Uuid;
+
+interface Message extends HasMessageName
+{
+    public const TYPE_COMMAND = 'command';
+    public const TYPE_EVENT = 'event';
+    public const TYPE_QUERY = 'query';
+
+    /**
+     * Should be one of Message::TYPE_COMMAND, Message::TYPE_EVENT or Message::TYPE_QUERY
+     */
+    public function messageType(): string;
+
+    public function uuid(): Uuid;
+
+    public function createdAt(): DateTimeImmutable;
+
+    public function payload(): array;
+
+    public function metadata(): array;
+
+    public function withMetadata(array $metadata): Message;
+
+    /**
+     * Returns new instance of message with $key => $value added to metadata
+     *
+     * Given value must have a scalar or array type.
+     */
+    public function withAddedMetadata(string $key, $value): Message;
+}
+```
+That is the basic contract, an immutable message with a unique identifier, a type, an created at timestamp, 
+a message name (by extending the `HasMessageName` interface), payload and metadata.
+Both payload and metadata should only contain scalar types and arrays (no objects) to be truly immutable.
```

Comment:
Note: The abstract class `DomainMessage` does only allow scalar values for metadata, not arrays!